src/test/java/org/broadinstitute/hellbender/tools/walkers/annotator/VariantAnnotatorIntegrationTest.java

```diff
@@ -1,342 +1,369 @@
-    }
-    @Test
-    public void testNoReads() throws IOException {
-        assertVariantContextsMatch(getTestFile("vcfexample3empty.vcf"), getTestFile("expected/testNoReads.vcf"), Arrays.asList("-G", "Standard", "-L", getToolTestDataDir() + "vcfexample3empty.vcf"), b37_reference_20_21);
-    }
-    @Test
-    public void testMultipleIdsWithDbsnp() throws IOException {
-        assertVariantContextsMatch(getTestFile("vcfdbsnpwithIDs.vcf"), getTestFile("expected/testMultipleIdsWithDbsnp.vcf"), Arrays.asList("-G", "Standard", "-L",getToolTestDataDir() + "vcfdbsnpwithIDs.vcf", "--dbsnp", dbsnp_138_b37_20_21_vcf), null);
-    }
-    @Test
-    public void testDBTagWithHapMap() throws IOException {
-        assertVariantContextsMatch(getTestFile("vcfexample3empty.vcf"),
-                getTestFile("expected/testDBTagWithHapMap.vcf"),
-                Arrays.asList("-G", "Standard", "-L", getToolTestDataDir() + "vcfexample3empty.vcf", "--comp:H3", getToolTestDataDir() + "fakeHM3.vcf"),
-                b37_reference_20_21, Collections.emptyList());
     }
 
-    // Specific results from this method have not been rigorously vetted, this test asserts that the annotations we expect are present (compared to gatk3 but not gatk3 output because of non-ported annotations )
+    // test dbSNP annotation
+    // the input VCF -- up to the ID field -- is as follows:
+    //  20	10002353	rs372143558     in dbSNP with same tag
+    //  20	10002374	.               not in dbSNP
+    //  20	10002443	rs1;rs2         in dbSNP with tag rs188831105
+    //  20	10002458	.               in dbSNP with tag rs34527371
+    //  20	10002460	rstestnot       not in dbSNP
+    //  20	10002470	rstest          in dbSNP with tag rs2327260
+    //  20	10002470	rstest          in dbSNP with tag rs3062214
+    //  20	10002478	.               in dbSNP with tag rs33961276
+    //  we expect the correct tag to be added to the ID field if necessary and the "DB" INFO tag to be added iff the variant is in dbSNP
     @Test
-    public void testWithAllAnnotations() throws IOException {
-        assertVariantContextsMatch(getTestFile("HCOutput.NoAnnotations.vcf"),
-                getTestFile("expected/testWithAllAnnotations.vcf"),
-                //TODO remove the -AX here when https://github.com/broadinstitute/gatk/issues/3944 is resolved
-                Arrays.asList("--"+ StandardArgumentDefinitions.ENABLE_ALL_ANNOTATIONS, "-AX", "ReferenceBases", "-L", "20:10000000-10100000", "-I", NA12878_20_21_WGS_bam),
-                 b37_reference_20_21);
+    public void testDBSNP() {
+        final File inputVCF = BASIC_INPUT;
+        final File outputVCF = createTempFile("output", ".vcf");
+
+        final ArgumentsBuilder args = new ArgumentsBuilder()
+                .addVCF(inputVCF)
+                .addOutput(outputVCF)
+                .addArgument(DbsnpArgumentCollection.DBSNP_LONG_NAME, dbsnp_138_b37_20_21_vcf);
```

Comment:
Are we supposed to match alleles with dbSNP now?  There certainly was a time when we didn't, i.e. a different ALT at the same position would still get annotated.

Reply:
That's unchanged.